test/extended/default_forcePull_test.go

```diff
@@ -0,0 +1,262 @@
+	s2iComp := s2iDockBldr + ":latest"
+	custComp := custBldr + ":latest"
+
+	comps := []string{s2iComp, custComp}
+	retIDs, gerr := exutil.GetImageIDForTags(comps)
+
+	Expect(gerr).NotTo(HaveOccurred())
+	By(fmt.Sprintf("\n%s******FORCE PULL TEST %s true ******:  compare image - %s, %s, %s, %s", time.Now().Format(time.RFC850), strategy, s2iComp, custComp, retIDs[0], retIDs[1]))
+	Expect(len(retIDs[0]) > 0).Should(BeTrue())
+	Expect(len(retIDs[1]) > 0).Should(BeTrue())
+	Expect(retIDs[0] != retIDs[1]).Should(BeTrue())
+}
+
+func waitForBuild(context string, oc *exutil.CLI) {
+	buildName := "ruby-sample-build-1"
+	By(fmt.Sprintf("\n%s******FORCE PULL TEST %s ******:   waiting for %s", time.Now().Format(time.RFC850), context, buildName))
+	exutil.WaitForABuild(oc.REST().Builds(oc.Namespace()), buildName,
+		// The build passed
+		func(b *buildapi.Build) bool {
+			return b.Name == buildName && b.Status.Phase == buildapi.BuildPhaseComplete
+		},
+		// The build failed
+		func(b *buildapi.Build) bool {
+			if b.Name != buildName {
+				return false
+			}
+			return b.Status.Phase == buildapi.BuildPhaseFailed || b.Status.Phase == buildapi.BuildPhaseError
+		},
+	)
+	// do not care if build returned an error ... entirely possible ... we only check if the image was updated or left the same appropriately
+	By(fmt.Sprintf("\n%s******FORCE PULL TEST %s  ******:   done waiting for %s", time.Now().Format(time.RFC850), context, buildName))
+}
+
+func doBuild(json string, oc *exutil.CLI) {
+	err := oc.Run("create").Args("-f", json).Execute()
+	Expect(err).NotTo(HaveOccurred())
+	_, berr := oc.Run("start-build").Args("ruby-sample-build").Output()
+	Expect(berr).NotTo(HaveOccurred())
+}
+
+var _ = BeforeSuite(func() {
+	// do a pull initially just to insure have the latest version
+	exutil.PullImage(s2iDockBldr)
+	exutil.PullImage(custBldr)
+	// save hex image IDs for image reset after corruption
+	tags := []string{s2iDockBldr + ":latest", custBldr + ":latest"}
+	hexIDs, ierr := exutil.GetImageIDForTags(tags)
+	Expect(ierr).NotTo(HaveOccurred())
+	for _, hexID := range hexIDs {
+		By(fmt.Sprintf("\n%s******FORCE PULL TEST ******:  hex id %s ", time.Now().Format(time.RFC850), hexID))
+	}
+	Expect(len(hexIDs)).To(Equal(2))
+	s2iStartingHexID = hexIDs[0]
+	custStartingHexID = hexIDs[1]
+	By(fmt.Sprintf("\n%s******FORCE PULL TEST ******:  hex id for s2i/docker %s and for custom %s ", time.Now().Format(time.RFC850), s2iStartingHexID, custStartingHexID))
+})
+
+var _ = Describe("\n******FORCE PULL TEST ******:  ForcePull from OpenShift induced builds (vs. sti)", func() {
+	defer GinkgoRecover()
+	Describe("\n******FORCE PULL TEST ******:  Force pull and s2i builder", func() {
+		// corrupt the s2i builder image
+		BeforeEach(func() {
+			corruptImage(s2iDockBldr, custBldr, "s21")
+		})
+
+		AfterEach(func() {
+			resetImage()
+		})
+
+		Context("\n******FORCE PULL TEST ******:  when s2i force pull is false and the image is bad", func() {
```

Comment:
I'll remove the ******************

I'm not as happy with the Ginko formatter as you are though ;-) ... I'm leaving the \n